packit_service/worker/events/event.py

```diff
@@ -379,29 +381,42 @@ def get_package_config(self) -> Optional[PackageConfig]:
             package_config.upstream_project_url = self.project_url
         return package_config
 
+    @staticmethod
+    def _filter_failed_models_target(
+        models: Union[
+            Optional[Iterable[CoprBuildModel]], Optional[Iterable[TFTTestRunModel]]
+        ],
+    ) -> Optional[Set[str]]:
+        failed_models_target = set()
+        for model in models:
+            if model.status in ["failed", "error"]:
+                failed_models_target.add(model.target)
+
+        return failed_models_target if failed_models_target else None
+
+    def get_all_tf_failed_targets(self) -> Optional[Set[str]]:
+        if self.commit_sha is None:
+            return None
+
+        return self._filter_failed_models_target(
+            models=TFTTestRunModel.get_all_by_commit_target(commit_sha=self.commit_sha)
+        )
+
+    def get_all_build_failed_targets(self) -> Optional[Set[str]]:
+        # TODO: get rid of project.repo
+        if self.commit_sha is None or self.project.repo is None:
+            return None
```

Comment:
Can you please make a comment about the `project.repo`? (So we know why that's here.)